tensorboard/webapp/widgets/line_chart_v2/lib/worker_pool.ts

```diff
@@ -0,0 +1,98 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {getWorker} from './worker';
+
+/**
+ * An object that that provides facility to interact with a Worker process. Because
+ * WorkerPool maintains lifecycle of a Worker, it does not permit `terminate` or any other
+ * direct binding to Worker events.
+ *
+ * For listening to events from Worker, please use MessageChannel pattern:
+ * https://developer.mozilla.org/en-US/docs/Web/API/MessageChannel
+ */
+export interface WorkerProxy {
+  activeCount: number;
+  postMessage: (message: any, transfer: Transferable[]) => void;
+  free: () => void;
+}
+
+/**
+ * Returns a worker-like object that can be used to offload computation like Worker.
```

Comment:
“Returns a worker-like object”—are these docs meant for `getNext` rather
than the class, or else should they be rephrased?